File: Leetcode/JS/Leetcode563-二叉树的坡度.js
var findTilt = function(root) {
    let ans = 0;
    const dfs = function(root){
        if(!root) return 0;
        let lhs = dfs(root.left),rhs = dfs(root.right);
        ans += Math.abs(lhs - rhs)
        return lhs + rhs + root.val;
    };
    dfs(root);
    return ans;
};

File: Leetcode/PY/Leetcode563-二叉树的坡度.py
class Solution:
    ans = 0
    def findTilt(self, root: TreeNode) -> int:
        def dfs(root):
            if not root: return 0
            lhs,rhs = dfs(root.left),dfs(root.right)
            self.ans += abs(lhs - rhs)
            return lhs + rhs + root.val
        dfs(root)
        return self.ans